Assets/GameAssets/Player/PlayerScripts/PlayerHealth.cs

```diff
@@ -43,8 +48,40 @@ public override void IncreaseHealth(float heal)
         base.IncreaseHealth(heal);
 
         PlayerManager.Instance.InvokePlayerHealthChangeEvent(GetHealthPercent(), _currentHealth);
+        CheckHeartBeat();
     }
 
+    private void CheckHeartBeat()
+    {
+        if (base._currentHealth <= _HeartBeatStart && base._currentHealth != _HeartBeatEnd && _heartBeatCoroutine == null)
```

Comment:
`base.` is unnecessary since its a protected variable. All derived classes can access the derived variable as normal!